lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -213,47 +233,32 @@ def get(self, request, *args, **kwargs):
                 'location': str(resume_block)
             })
             resume_course['url'] = request.build_absolute_uri(resume_path)
+        elif allow_public_outline or allow_public:
+            course_blocks = get_course_outline_block_tree(request, course_key_string, None)
```

Comment:
Stuff like this duplicated line is why I had this view do the 'per-item' approach vs the 'if-block' approach. Not all our item's conditions map cleanly to simple if-blocks, so you end up duplicating some code.

The old LMS view felt like spaghetti code to me because of this and it felt easy for a future person to miss that they should also fix the rarely-used public-outline code path etc. So I tried the per-item approach.

But I don't disagree that it became unwieldy too. Would breaking up this method would add clarity? Probably not.

I'm down with your redesign. The current amount of duplicated code is very small.